server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1293,6 +1295,33 @@ public static Setting<Float> floatSetting(
         );
     }
 
+    private class FloatParser implements Function<String, Float>, Writeable {
+        private String key;
+        private float minValue;
+        private float maxValue;
+        private boolean isFiltered;
+
+        public FloatParser(String key, float minValue, float maxValue, boolean isFiltered) {
```

Comment:
(Style) For readability, it's best to keep the parameters here in the same order as the `parseFloat()` method below.